### DISCUSSION TITLE
[Core Protocol] Example of how to upgrade Proposer or Challenger addresses in L2OutputOracle.sol?

### DISCUSSION THREAD

**Discussion by RnkSngh** [ethereum-optimism/developers#258] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I was looking through the [L2OutputOracle.sol](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/L2OutputOracle.sol) and on [this line](https://github.com/ethereum-optimism/optimism/blob/416cdccd2a93f83fe713dea3b0d86889c430dce3/packages/contracts-bedrock/src/L1/L2OutputOracle.sol#L36) the comment says the address of the Proposer can be updated via upgrade. It also says something similar for the Challenger. 

I was curious how an example of updating either address would look like? Is it basically upgrading to a new L2OutputOracle logic contract that has a setter function to set a new Proposer/Challenger? If so, why is this comment only present on the Challenger and Proposer variables (presumably you can do the same to all other variables stored in the contract) ?

**Comment by opfocus** [comment]
https://github.com/ethereum-optimism/optimism/blob/a06cae81ed60c85d1566f48f45d9f9c0e48cc47c/packages/contracts-bedrock/scripts/getting-started/config.sh#L48      Here we can see pre-deployed contract configuration information.       https://etherscan.io/address/0xdb5d932af15d00f879cabebf008cadaaaa691e06#code   In the logic contract, these pieces of information are initialized (including other information not mentioned in the comments). I believe upgrading them would require deploying a new logic contract.       Then, the address of the executing contract should be updated in the proxy contract.

**Reply by sbvegan** [reply]
This is correct, we don't have any public documentation on this yet, but I'll create an issue on the docs repo. You'd accomplish this with the proxy admin wallet. [Docs on privileged roles](https://docs.optimism.io/chain/security/privileged-roles)
